internal/resources/attributes.go

```diff
@@ -28,6 +28,34 @@ const (
 	deprecationMessage = "Remove this attribute's configuration as it no longer in use and the attribute will be removed in the next major version of the provider."
 )
 
+// WithDescription sets the MarkdownDescription for the provided attribute.
+// It accepts an attribute of type schema.StringAttribute, schema.Int64Attribute,
+// schema.BoolAttribute, schema.SetAttribute, schema.Float64Attribute,
+// schema.NumberAttribute, or schema.ListAttribute, and a description string.
+// The function returns the modified attribute.
+func WithDescription[T interface {
+	*schema.StringAttribute | *schema.Int64Attribute | *schema.BoolAttribute | *schema.SetAttribute |
+		*schema.Float64Attribute | *schema.NumberAttribute | *schema.ListAttribute
+}](attr T, description string) T {
+	switch v := any(attr).(type) {
```

Comment:
since we're doing type switch, don't think generics is a good fit here.  

I just found with go generics we can't access struct fields. see https://github.com/golang/go/issues/48522.

seems like we should use interface instead.

Reply:
never mind just saw Michael's comment, so that is valid approach

Reply:
ya its obv not ideal but it does force a caller to only use the types we want which is good. maybe sometime in the future go will support field restriction on types like in typescript where u can say hey accept a type as an arg that has a certain field